docs/connector-development/config-based/pagination.md

```diff
@@ -0,0 +1,100 @@
+# HTTP Requester
+
+Given a page size and a pagination strategy, the `LimitPaginator` will point to pages of results for as long as its strategy returns a `next_page_token`.
+
+Iterating over pages of result is different from iterating over stream slices.
+Stream slices have semantic value, for instance, a Datetime stream slice defines data for a specific date range. 2 stream slices will have data for different date ranges.
+Conversely, pages don't have semantic value. More pages simply means that more records are to be read, without specifying any meaningful difference between the records of the first and later pages.
+
+The paginator is defined by
+
+- page size: the number of records to fetch in a single request
+- limit_option: how to specify the page size in the outgoing HTTP request
+- pagination_strategy: how to compute the next page to fetch
+- page_token_option: how to specify the next page to fetch in the outgoing HTTP request
+
+3 pagination strategies are supported
+
+1. Page increment
+2. Offset increment
+3. Cursor-based
+
+## Pagination Strategies
+
+### Page increment
+
+When using the `PageIncrement` strategy, the page number will be set as part of the `page_token_option`.
+
+The following paginator example will fetch 5 records per page, and specify the page number as a request_parameter:
+
+```
+paginator:
+  type: "LimitPaginator"
+  page_size: 5
+  limit_option:
+    option_type: request_parameter
+    field_name: page_size
+  pagination_strategy:
+    type: "PageIncrement"
+  page_token:
+    option_type: "request_parameter"
+    field_name: "page"
+```
+
```

Comment:
General suggestion on snippets you share:
add an example of the corresponding generated HTTP requests. I think it could help the developer familiar with REST API quickly grasp the concepts you try to explain and map these to their use cases.

Reply:
added examples